servo/servo.go

```diff
@@ -80,3 +86,17 @@ func (s Servo) SetMicroseconds(microseconds int16) {
 	value := uint64(s.pwm.Top()) * uint64(microseconds) / (pwmPeriod / 1000)
 	s.pwm.Set(s.channel, uint32(value))
 }
+
+// SetAngle sets the angle of the servo in degrees. The angle should be between
+// 0 and 180, where 0 is the minimum angle and 180 is the maximum angle.
+func (s Servo) SetAngle(angle int) error {
+	if angle < 0 || angle > 180 {
+		return ErrInvalidAngle
+	}
+
+	// 0° is 1000µs, 180° is 2000µs
+	microseconds := angle*1000/180 + 1000
+	s.SetMicroseconds(int16(microseconds))
```

Comment:
Doesn't this vary between servo motors?
I think this needs some explanation that it varies by servo motor. For example, from Wikipedia:

> The parameters for the pulses are the minimal pulse width, the maximal pulse width, and the repetition rate. Given the rotation constraints of the servo, neutral is defined to be the center of rotation. Different servos will have different constraints on their rotation, but the neutral position is always around 1.5 milliseconds (ms) pulse width.

Reply:
I added specific reference to `SetMicroseconds()` and some additional info as well. Should be good now.